### PR TITLE
Addedd Unit Test to EmailType And EmailTransaction Entities

### PR DESCRIPTION
**Pull request by malamadhu**
UnitTest To EmailType and EmailTransaction

### CODE REVIEW

File: BookingSystem.DAL/Repository/EmailTransactionRepository.cs [hunk 1]
```diff
@@ -29,10 +29,9 @@ public async Task<bool> SaveEmailTransactionAsync(EmailTransactionVM transaction
       emailTransaction.EmailTypeID = transactionVM.EmailTypeID;
       emailTransaction.Send_Status = transactionVM.Send_Status;
       _dataContext.EmailTransactions.Add(emailTransaction);
-      if (await _dataContext.SaveChangesAsync() > 0)
-        return true;
-      else
-        return false;
+      await _dataContext.SaveChangesAsync();
```
Comment (Backiaraj07) [inline BookingSystem.DAL/Repository/EmailTransactionRepository.cs:32]: Use ConfigureAwait(false) at the end

File: BookingSystem.DAL/Repository/EmailTypeRepository.cs [hunk 2]
```diff
@@ -28,11 +28,11 @@ public async Task<bool> SaveEmailTypeAsync(EmailTypeVM model)
             emailType.Subject = model.Subject;
             emailType.IsDeleted = false;
             _dataContext.EmailTypes.Add(emailType);
-            if (await _dataContext.SaveChangesAsync() > 0)
-                return true;
-            else 
-                return false;
+            await _dataContext.SaveChangesAsync();
```
Comment (Backiaraj07) [inline BookingSystem.DAL/Repository/EmailTypeRepository.cs:31]: Use ConfigureAwait(false) at the end

File: BookingSystem.Tests/APIController.Tests/APIEmailTransactionController.Tests/SaveEmailTransactionTest.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      _mockLog.Setup(m => m.Log(LogLevel.Information, It.IsAny<EventId>(), It.IsAny<object>(), It.IsAny<Exception>(), It.IsAny<Func<object, Exception, string>>()));
+
+      var mockLoggerFactory = new Mock<ILoggerFactory>();
+      mockLoggerFactory.Setup(x => x.CreateLogger(It.IsAny<string>())).Returns(() => _mockLog.Object);
+
+      _mockEmailTransactionService.Setup(repo => repo.SaveEmailTransactionAsync(It.IsAny<EmailTransactionVM>())).Throws(new Exception());
+      var _emailTransactionController = new EmailTransactionController(_mockEmailTransactionService.Object, mockLoggerFactory.Object);
+
+      //Act
+      var exception = await _emailTransactionController.SaveEmailTransactionAsync(It.IsAny<EmailTransactionVM>());
```
Comment (Backiaraj07) [inline BookingSystem.Tests/APIController.Tests/APIEmailTransactionController.Tests/SaveEmailTransactionTest.cs:71]: Use ConfigureAwait(false) at the end

File: BookingSystem.Tests/APIController.Tests/APIEmailTypeController.Tests/SaveEmailTypeTest.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      _mockLog.Setup(m => m.Log(LogLevel.Information, It.IsAny<EventId>(), It.IsAny<object>(), It.IsAny<Exception>(), It.IsAny<Func<object, Exception, string>>()));
+
+      var mockLoggerFactory = new Mock<ILoggerFactory>();
+      mockLoggerFactory.Setup(x => x.CreateLogger(It.IsAny<string>())).Returns(() => _mockLog.Object);
+
+      _mockEmailTypeService.Setup(repo => repo.SaveEmailTypeAsync(It.IsAny<EmailTypeVM>())).Throws(new Exception());
+      var _emailTypeController = new EmailTypeController(_mockEmailTypeService.Object, mockLoggerFactory.Object);
+
+      //Act
+      var exception = await _emailTypeController.SaveEmailTypeAsync(It.IsAny<EmailTypeVM>());
```
Comment (Backiaraj07) [inline BookingSystem.Tests/APIController.Tests/APIEmailTypeController.Tests/SaveEmailTypeTest.cs:73]: Use ConfigureAwait(false) at the end

File: BookingSystem.Tests/BAL.Tests/EmailTransactionServiceTest/SaveEmailTransactionTest.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    public async Task SaveEmailTransaction_ReturnOKResult_WithEmailTransactionlId()
+    {
+      // Arrange
+      var data = fixture.Build<EmailTransactionVM>().With(x => x.EmailId).Create();
+      _emailTransactionRepository.Setup(repo => repo.SaveEmailTransactionAsync(data));
+      var expectedemailtypeId = data.EmailId;
+      var _emailTransactionService = new EmailTransactionService(_emailTransactionRepository.Object);
+
+      // Act
+      var result = await _emailTransactionService.SaveEmailTransactionAsync(data);
```
Comment (Backiaraj07) [inline BookingSystem.Tests/BAL.Tests/EmailTransactionServiceTest/SaveEmailTransactionTest.cs:35]: Use ConfigureAwait(false) at the end

File: BookingSystem.Tests/BAL.Tests/EmailTypeServiceTest/SaveEmailTypeTest.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    public async Task SaveEmailType_ReturnOKResult_WithEmailTypelId()
+    {
+      // Arrange
+      var data = fixture.Build<EmailTypeVM>().With(x => x.EmailTypeID).Create();
+      _emailTypeRepository.Setup(repo => repo.SaveEmailTypeAsync(data));
+      var expectedemailtypeId = data.EmailTypeID;
+      var _emailtypeService = new EmailTypeService(_emailTypeRepository.Object);
+
+      // Act
+      var result =await _emailtypeService.SaveEmailTypeAsync(data);
```
Comment (Backiaraj07) [inline BookingSystem.Tests/BAL.Tests/EmailTypeServiceTest/SaveEmailTypeTest.cs:36]: Use ConfigureAwait(false) at the end

File: BookingSystem.Tests/Integration.Tests/EmailTransactionRepositoryTest/SaveEmailTransactionTest.cs [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+      //create In Memory Database
+      var options = new DbContextOptionsBuilder<DataContext>().UseInMemoryDatabase(dbconnection).Options;
+      using (var context = new DataContext(options))
+      {
+        context.EmailTransactions.RemoveRange(context.EmailTransactions);
+        _emailTransactionRepository = new EmailTransactionRepository(context);
+
+        // Act
+        var result = await _emailTransactionRepository.SaveEmailTransactionAsync(emailTransactionData);
```
Comment (Backiaraj07) [inline BookingSystem.Tests/Integration.Tests/EmailTransactionRepositoryTest/SaveEmailTransactionTest.cs:68]: Use ConfigureAwait(false) at the end

**Review by Backiaraj07 — COMMENTED** [review]
Comments added

**Review by Backiaraj07 — APPROVED** [review]
fixed

**Review by manic1990 — APPROVED** [review]

**Pull request merged**